Fabric.Authorization.Domain/Stores/Services/PermissionService.cs

```diff
@@ -139,6 +133,32 @@ public PermissionService(IPermissionStore permissionStore, RoleService roleServi
         }
 
         /// <summary>
+        /// removes granular permissions from a user
+        /// </summary>
+        /// <param name="granularPermission"></param>
+        /// <returns></returns>
+        public async Task DeleteGranularPermissions(GranularPermission granularPermission)
+        {
+            try
+            {
+                var stored = await GetUserGranularPermissions(granularPermission.Id);
+
+                var additionalPermissionsToRemove = granularPermission.AdditionalPermissions.Select(p => p.ToString()).ToList();
+                stored.AdditionalPermissions = stored.AdditionalPermissions.Where(p => !additionalPermissionsToRemove.Contains(p.ToString()));
+
+                var deniedPermissionsToRemove = granularPermission.DeniedPermissions.Select(ap => ap.ToString()).ToList();
+                stored.DeniedPermissions = stored.DeniedPermissions.Where(p => !deniedPermissionsToRemove.Contains(p.ToString()));
+
+                await _permissionStore.AddOrUpdateGranularPermission(stored);
+
+            }
+            catch (NotFoundException<GranularPermission>)
+            {
+                // Do nothing
```

Comment:
Is there a reason we swallow this exception? Should we log the exception?

Reply:
i did not think it seemed important enough to log. if the user has had no granular permissions added then there is nothing to delete and so not much to do.

Reply:
I think it might be better to allow this exception to bubble up, and catch it in the module, and we can return a BadRequest with the appropriate error message so the consumer of the API knows that the user doesn't have any granular permissions defined.

Reply:
sure thing i can do that. BadRequest doesnt feel like the proper response code but I am having trouble thinking of a better one.

Reply:
Yes, sorry, 404 would be better.